src/main/java/com/google/sps/data/User.java

```diff
@@ -18,6 +29,43 @@ public static boolean createdRecipe(String userId, String recipeId) {
     Boolean userCreatedRecipe =
         user.getBoolean(DBUtils.getNestedPropertyName(CREATED_RECIPES_KEY, recipeId));
     // note that userCreatedRecipe is a Boolean, not a boolean
-    return userCreatedRecipe != null && userCreatedRecipe == true;
+    return userCreatedRecipe != null && userCreatedRecipe;
+  }
+
+  public static boolean isSavedRecipe(String userId, String recipeId) {
```

Comment:
This should not be static. Use the existing `User.uid` property instead of taking in a new `userId`.

Reply:
I think I'm gonna revert User.java to it's master state and just check for saved and created in my own functions

Reply:
It would be good to add those methods in the interface as well.

Reply:
Ok, will do